src/Thujohn/Twitter/Traits/AccountActivityTrait.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+namespace Thujohn\Twitter\Traits;
+
+use BadMethodCallException;
+
+trait AccountActivityTrait
+{
+	/**
+	 * Creates HMAC SHA-256 hash from incomming crc_token and consumer secret.
+	 * This base64 encoded hash needs to be returned by the application when Twitter calls the webhook.
+	 *
+	 * @param  mixed $crcToken
+	 *
+	 * @return void
+	 */
+	public function CrcHash($crcToken)
+	{
+		$hash = hash_hmac('sha256', $crcToken, $this->tconfig['CONSUMER_SECRET'], true);
+
+		return 'sha256='.base64_encode($hash);
+	}
+	
+	/**
+	 * Registers a webhook $url for all event types in the given environment.
+	 *
+	 * @param  mixed $env
+	 * @param  mixed $url
+	 *
+	 * @return void
+	 */
+	public function setWebhook($env, $url)
+	{
+		return $this->post("account_activity/all/{$env}/webhooks", ['url' => $url]);
+	}
+
+	/**
+	 * Returns webhook URLs for the given environment (or all environments if none provided), and their statuses for the authenticating app. 
+	 *
+	 * @param  mixed $env
+	 *
+	 * @return void
+	 */
+	public function getWebhooks($env = null)
+	{
+		return $this->get("account_activity/all/".($env ? $env."/" : ""). "webhooks");
+	}
+
+	/**
+	 * Triggers the challenge response check (CRC) for the given enviroments webhook for all activites.
+	 * If the check is successful, returns 204 and reenables the webhook by setting its status to valid.
+	 *
+	 * @param  mixed $env
+	 * @param  mixed $webhookId
+	 *
+	 * @return void
+	 */
+	public function updateWebhooks($env, $webhookId)
+	{
+		$this->query("account_activity/all/{$env}/webhooks/{$webhookId}", "PUT");
+
+		return "HTTP {$this->response['code']} {$this->response['response']}";
```

Comment:
can we return an array/object here rather than a string please?

Reply:
(and below)

Reply:
You mean the whole `$this->response`? It's kinda huge. All the requests which are returning the status code + body (string) are HTTP 204 on success. The default behavior returns the response body, but in this case it's empty because it doesn't have a body. Perhaps return boolean or status code (int) for these? If the status has errors, `RunTimeException` will kick in anyway. What do you think?

Reply:
I was suggesting returning an array with the code, but I agree it would be best to return a bool/status code.

I think bool if probably safe, or if caught by runtime exception just not return anything?

Reply:
Yeah, I think bool will be suitable here. The runtime exception is thrown in the main `query()` method, which returns the code and response body.